samples/cpp/image_generation/README.md

```diff
@@ -148,3 +149,59 @@ The resuling image is:
    ![](./inpainting.bmp)
 
 Note, that LoRA, heterogeneous execution and other features of `Text2ImagePipeline` are applicable for `InpaintingPipeline`.
+
+## benchmarking sample for text to image pipeline
+
+This `benchmark_text2image.cpp` sample script demonstrates how to benchmark the text to image pipeline, image to image pipeline and inpainting pipeline. The script includes functionality for warm-up iterations, generating image, and calculating various performance metrics.
+
+The usage of this sample is:
+```bash
+./benchmark_text2image [OPTIONS]
+```
+Options:
+- `-pt, --pipeline_type` (default: `"text2image"`): Pipeline type(text2image, image2image, inpainting).
+- `-m, --model`: Path to the model and tokenizers base directory.
+- `-p, --prompt` (default: `"The Sky is blue because"`): The prompt to generate text.
+- `-nw, --num_warmup` (default: `1`): Number of warmup iterations.
+- `-n, --num_iter` (default: `3`): Number of iterations.
+- `-d, --device` (default: `"CPU"`): Device to run the model on.
+- `-wh, --width` (default: `512`): The width of the output image.
+- `-ht, --height` (default: `512`): The height of the output image.
+- `-is, --num_inference_steps` (default: `20`): The number of inference steps.
+- `-ni, --num_images_per_prompt` (default: `1`): The number of images to generate per generate() call.
+- `-o, --output_dir` (default: `""`): Path to save output image.
+- `-i, --image`: Path to input image.
+- `-s, --strength`: Indicates extent to transform the reference `image`. Must be between 0 and 1.
+- `-mi, --mask_image`: Path to mask image.
+
+For example:
+
+`./benchmark_text2image -pt text2image -m dreamlike_anime_1_0_ov/FP16 -n 10 -d CPU`
+
+Performance output:
+
+```
+[warmup-0] generate time: 4978.00 ms, total infer time:4967.62 ms, infer step num:20
+[warmup-0] encoder infer time: (text_encoder-195.93 ms) (text_encoder_2-145.52 ms)
+[warmup-0] unet total infer time:3920.33 ms, infer number:20
```

Comment:
could you please add `iteration duration` for transformer and Unet ?

It would be great to see difference between first and other iterations.